app/controllers/conversations_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ConversationsController < ApplicationController
+
+  def show
+    @message = Message.new
+    @conversation = Conversation.find(params[:id])
+    @messages = @conversation.messages.order(:created_at)
+  end
+
+  def recent
+    @recent_conversations = Conversation.recently_active(25)
+  end
+
+  def create
+    @conversation = Conversation.new(conversation_params)
```

Comment:
In your `conversation_params` method you could chain on:

```ruby
.merge(author: current_user, topic: Topic.find(params[:topic_id])
```
... then you wouldn't need lines 15 and 16.